plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1710,6 +1721,8 @@ private function get_diff_meta_data_between_orders( \WC_Abstract_Order &$order1,
 	 * @return void
 	 */
 	private function migrate_post_record( \WC_Abstract_Order &$order, \WC_Abstract_Order $post_order ): void {
+		self::$sync_on_read_order_ids[] = $order->get_id();
```

Comment:
It's a small thing, but for performance, should we make this a map and check `isset()` instead?